### PR TITLE
Fixing workingDirectory not honored for .NET Core task build 

### PR DESCRIPTION
**Pull request by issacnitin**
Fixing issues https://github.com/microsoft/azure-pipelines-tasks/issues/12717 and https://github.com/microsoft/azure-pipelines-tasks/issues/10996 for dotnet restore and build

Root cause is that while searching for project files, workingDirectory input is not considered

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
@issacnitin can please link the issue and a short description of the root cause and the fix

**Comment by bishal-pdMSFT** [comment]
@issacnitin I am not completely sure whether to need to "fix" this. IMO this is not really an issue. Minimatch pattern in `projects` input will allow user to select projects under a specific directory. The `workingDirectory` input is not required for that. 
`workingDirectory` is used when running `dotnet.exe` commands. Ans that is being honored correctly.

Also, IMO this is a breaking change and it might start breaking existing customers.

**Comment by bishal-pdMSFT** [comment]
@tejasd1990 your opinion please!

**Comment by issacnitin** [comment]
@bishal-pdMSFT If you feel this is not an issue, please close the associated github items. I’m surprised the items has been open for so long no action was taken to close it as it don’t need the fix.

Yes this is potentially a breaking change

File: Tasks/DotNetCoreCLIV2/task.loc.json [hunk 1]
```diff
@@ -490,7 +490,8 @@
       "label": "ms-resource:loc.input.label.workingDirectory",
       "helpMarkDown": "ms-resource:loc.input.help.workingDirectory",
       "required": "false",
-      "groupName": "generalAdvanced"
+      "groupName": "generalAdvanced",
+      "visibleRule": "command = custom || command = build"
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/task.loc.json:494]: WD is used for all non package related commands.

```
case "build":
                case "publish":
                case "run":
                    await this.executeBasicCommand();
                    break;
                case "custom":
                    this.command = tl.getInput("custom", true);
                    await this.executeBasicCommand();
                    break;
                case "test":
                    await this.executeTestCommand();
                    break;
```

File: Tasks/DotNetCoreCLIV2/task.loc.json [hunk 2]
```diff
@@ -490,7 +490,8 @@
       "label": "ms-resource:loc.input.label.workingDirectory",
       "helpMarkDown": "ms-resource:loc.input.help.workingDirectory",
       "required": "false",
-      "groupName": "generalAdvanced"
+      "groupName": "generalAdvanced",
+      "visibleRule": "command != pack"
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/task.loc.json:494]: shouldn't this be restore and push as well

File: Tasks/DotNetCoreCLIV2/task.json [hunk 3]
```diff
@@ -488,9 +488,10 @@
             "name": "workingDirectory",
             "type": "filePath",
             "label": "Working directory",
-            "helpMarkDown": "Current working directory where the script is run. Empty is the root of the repo (build) or artifacts (release), which is $(System.DefaultWorkingDirectory)",
+            "helpMarkDown": "Current working directory where the script is run. Empty is the root of the repo (build) or artifacts (release), which is $(System.DefaultWorkingDirectory). Supported only for 'build' and 'custom' command. In case of 'build', project search pattern is **NOT** relative to working directory.",
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/task.json:491]: this also needs to change

File: Tasks/DotNetCoreCLIV2/Strings/resources.resjson/en-US/resources.resjson [hunk 4]
```diff
@@ -81,8 +81,8 @@
   "loc.input.help.buildProperties": "Specifies a list of token = value pairs, separated by semicolons, where each occurrence of $token$ in the .nuspec file will be replaced with the given value. Values can be strings in quotation marks.",
   "loc.input.label.verbosityPack": "Verbosity",
   "loc.input.help.verbosityPack": "Specifies the amount of detail displayed in the output.",
-  "loc.input.label.workingDirectory": "Working Directory",
-  "loc.input.help.workingDirectory": "Current working directory where the script is run. Empty is the root of the repo (build) or artifacts (release), which is $(System.DefaultWorkingDirectory)",
+  "loc.input.label.workingDirectory": "Working directory",
+  "loc.input.help.workingDirectory": "Current working directory where the script is run. Empty is the root of the repo (build) or artifacts (release), which is $(System.DefaultWorkingDirectory). Supported only for 'build' and 'custom' command. In case of 'build', project search pattern is **NOT** relative to working directory.",
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreCLIV2/Strings/resources.resjson/en-US/resources.resjson:85]: I think this will get updated when you build the code

**Comment by mhudasch** [comment]
Hi, I just lost multiple days in development time because the documentation of the task does not mention the behavior of the projects matcher ignoring the working directory. It has very little to no use to only fix or extend the help tooltip of this problem when one is not using the settings ui of this task. A part of being able to edit the build pipeline using yaml in any editor is that this help is not available especially when one is editing the task deep down in another repository inside a pipeline template. 
I would really urge to update the online documentation of this task and close all related issues like #9789 and #10996 oand clearly communicate your intentions with the *workingDirectory* parameter. This [shrug](https://github.com/microsoft/azure-pipelines-tasks/pull/13010#issuecomment-636145153) of a comment is the only place that i could find (including sifting through the codebase) that mentions the current stance. Sorry for the rant but this was really frustrating.

**Pull request merged**